File: src/com/ptby/dynamicreturntypeplugin/config/valuereplacement/MaskValueReplacementStrategy.java
package com.ptby.dynamicreturntypeplugin.config.valuereplacement;

import com.ptby.dynamicreturntypeplugin.index.ClassConstantWalker;

public class MaskValueReplacementStrategy implements ValueReplacementStrategy {

    private final String mask;


    public MaskValueReplacementStrategy( String mask ) {
        this.mask = mask;
    }


    @Override
    public String toString() {
        return "MaskValueReplacementStrategy{" +
                "\nmask='" + mask + '\'' +
                '}';
    }


    @Override
    public boolean equals( Object o ) {
        if ( this == o ) {
            return true;
        }
        if ( !( o instanceof MaskValueReplacementStrategy ) ) {
            return false;
        }

        MaskValueReplacementStrategy that = ( MaskValueReplacementStrategy ) o;

        //noinspection RedundantIfStatement
        if ( mask != null ? !mask.equals( that.mask ) : that.mask != null ) {
            return false;
        }

        return true;
    }


    @Override
    public int hashCode() {
        return mask != null ? mask.hashCode() : 0;
    }


    @Override
    public String replaceCalculatedValue( String currentValue ) {
        return String.format(
                mask, currentValue
        );
    }
}

File: src/com/ptby/dynamicreturntypeplugin/config/valuereplacement/MaskValueReplacementStrategy.kt
package com.ptby.dynamicreturntypeplugin.config.valuereplacement

import com.intellij.openapi.project.Project
import com.ptby.dynamicreturntypeplugin.signature_extension.isPhpClassConstantSignature
import com.ptby.dynamicreturntypeplugin.signature_extension.stripPhpClassConstantReference

public class MaskValueReplacementStrategy(private val mask: String) : ValueReplacementStrategy {


    override fun toString(): String {
        return "MaskValueReplacementStrategy{\nmask='$mask'}"
    }


    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is MaskValueReplacementStrategy) {
            return false
        }

        return mask == o.mask
    }


    override fun hashCode(): Int {
        return mask.hashCode()
    }


    override fun replaceCalculatedValue(project : Project, currentValue: String): String {
        val replacableValue = if ( currentValue.isPhpClassConstantSignature() ) {
            currentValue.stripPhpClassConstantReference()
        } else {
            currentValue
        }
        return mask.format(replacableValue)
    }
}
